ui/src/features/Editor/components/Canvas/useCreateNode.ts

```diff
@@ -0,0 +1,98 @@
+import { XYPosition } from "@xyflow/react";
+
+import { config } from "@flow/config";
+import { fetcher } from "@flow/lib/fetch/transformers/useFetch";
+import { nodeTypes, type Action, type Node, type NodeType } from "@flow/types";
+import { randomID } from "@flow/utils";
+
+import { baseBatchNode } from "./components/Nodes/BatchNode";
+import { baseNoteNode } from "./components/Nodes/NoteNode";
+
+type CreateNodeOptions = {
+  position: XYPosition;
+  type: string;
+  action?: Action;
+};
+
+const createBaseNode = ({ position, type }: CreateNodeOptions): Node => ({
+  id: randomID(),
+  position,
+  type,
+  data: {
+    name: type,
+    status: "idle",
+    locked: false,
+  },
+});
+
+const createSpecializedNode = ({
+  position,
+  type,
+}: CreateNodeOptions): Node | null => {
+  const node = createBaseNode({ position, type });
+
+  switch (type) {
+    case "batch":
+      return { ...node, ...baseBatchNode };
+    case "note":
+      return {
+        ...node,
+        data: { ...node.data, ...baseNoteNode },
+      };
+    default:
+      return null;
+  }
+};
+
+const createNodeFromAction = (action: Action, position: XYPosition): Node => ({
+  ...createBaseNode({ position, type: action.type }),
+  // Needs measured, but at time of creation we don't know size yet.
```

Comment:
Can you consolidate params. For two you are passing an object, and then for createNodeFromAction you are passing individual params 🙏🏻